android/src/main/java/com/pdftron/reactnative/views/DocumentView.java

```diff
@@ -1698,6 +1703,110 @@ public boolean handleBackButton() {
         return false;
     }
 
+    public void setPropertyForAnnotation(String annotId, int pageNumber, ReadableMap propertyMap) throws PDFNetException {
+        PDFViewCtrl pdfViewCtrl = getPdfViewCtrl();
+
+        boolean shouldUnlock = false;
+        try {
+            pdfViewCtrl.docLock(true);
+            shouldUnlock = true;
+
+            Annot annot = ViewerUtils.getAnnotById(pdfViewCtrl, annotId, pageNumber);
+            if (annot != null && annot.isValid()) {
+
+                if (propertyMap.hasKey(KEY_annotContents)) {
+                    String contents = propertyMap.getString(KEY_annotContents);
+                    if (contents != null) {
+                        annot.setContents(contents);
+                    }
+                }
+
+                if (propertyMap.hasKey(KEY_annotRect)) {
+                    ReadableMap rectMap = propertyMap.getMap(KEY_annotRect);
+                    if (rectMap != null) {
+                        double rectX1 = 0, rectY1 = 0, rectX2 = 0, rectY2 = 0;
+
+                        // to count how many keys exist
+                        int successCounter = 0;
+                        if (rectMap.hasKey(KEY_x1)) {
```

Comment:
use and operation with 4 condition instead of counter for better code readability